server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java

```diff
@@ -911,6 +912,49 @@ private void createSnapshotWithTablesAndMetadata() throws Exception {
         execute("DROP FUNCTION custom(string)");
     }
 
+    @Test
+    @UseRandomizedSchema(random = false)
+    public void test_restore_with_prefix_non_partitioned_table() throws Exception {
+        createTable("my_table_1", false);
+        createTable("my_table_2", false);
+        createSnapshot(SNAPSHOT_NAME, "my_table_1", "my_table_2");
+        waitNoPendingTasksOnAll();
+        execute("DROP TABLE my_table_1");
+
+        execute("RESTORE SNAPSHOT " + snapshotName() + " TABLE my_table_1 with (" +
+            "wait_for_completion=true," +
+            "postfix = 'test_postfix')"
+        );
+
+        execute("select table_name from information_schema.tables " +
+            "where table_name like 'my_table%' " +
+            "order by table_name"
+        );
+        assertThat(response).hasRows("my_table_1_test_postfix", "my_table_2");
+    }
+
+    @Test
+    @UseRandomizedSchema(random = false)
+    public void test_restore_with_prefix_partitioned_table() throws Exception {
+        createTable("my_parted_table_1", true);
+        createTable("my_parted_table_2", true);
+        createSnapshot(SNAPSHOT_NAME, "my_parted_table_1", "my_parted_table_2");
+        waitNoPendingTasksOnAll();
+        execute("DROP TABLE my_parted_table_1");
+        execute("DROP TABLE my_parted_table_2");
+
+        execute("RESTORE SNAPSHOT " + snapshotName() + " ALL with (" +
+            "wait_for_completion=true," +
+            "postfix = 'test_postfix')"
+        );
+
+        execute("select table_name from information_schema.tables " +
+            "where table_name like 'my_parted_table%' " +
+            "order by table_name"
+        );
+        assertThat(response).hasRows("my_parted_table_1_test_postfix", "my_parted_table_2_test_postfix");
```

Comment:
TODO: Handle template renaming (index renaming is already supported by ES but templates looks like not)